pyqir/src/qis.rs

```diff
@@ -12,309 +12,408 @@ use llvm_sys::{
 use pyo3::prelude::*;
 use qirlib::qis;
 
-/// An instruction builder that generates instructions from the basic quantum instruction set.
+/// Inserts a barrier instruction
 ///
-/// :param Builder builder: The underlying builder used to build QIS instructions.
-#[pyclass]
+/// :param builder: The underlying builder used to build QIS instructions.
+/// :rtype: None
+#[pyfunction]
 #[pyo3(text_signature = "(builder)")]
-pub(crate) struct BasicQisBuilder {
+#[allow(clippy::needless_pass_by_value)]
+pub(crate) fn barrier(py: Python, builder: Py<Builder>) {
```

Comment:
For all of these, you should be able to take in a `builder: &Builder`, which saves you needing to borrow it explicitly.